src/manifest/ManifestApi.ts

```diff
@@ -9,11 +10,42 @@ import { Scopes } from './api/Scopes';
  * @class ManifestApi
  */
 export class ManifestApi {
+  /**
+   * The list of scopes will be used for Outreach API authentication
+   * where current Outreach user will be asked to consent for granting
+   * permissions for defined scopes to addon creator.
+   *
+   * @see https://github.com/getoutreach/extensibility-sdk/blob/master/docs/manifest.md#scopes
+   * @type {Scopes[]}
+   * @memberof ManifestApi
+   */
+  scopes: Scopes[];
+
+  /**
+   *
+   * Outreach API client data used for authentication
+   *
+   * @see https://github.com/getoutreach/extensibility-sdk/blob/master/docs/manifest.md#client
+   * @type {ManifestApiClient}
+   * @memberof ManifestApi
+   */
+  client: ManifestApiClient;
+
+  /**
+   * Outreach OAuth App redirect uris on which the Authorization endpoint is implemented.
+   *
+   * @see https://github.com/getoutreach/extensibility-sdk/blob/master/docs/manifest.md#redirecturis
+   * @type {string}
+   * @memberof ManifestApi
+   */
+  redirectUris: string[];
+
   /**
    * Outreach OAuth application id
    *
    * @see https://github.com/getoutreach/extensibility-sdk/blob/master/docs/manifest.md#applicationid
    * @type {string}
+   * @deprecated applicationId is deprecated and replaced by client.id
```

Comment:
Why not just removing this? 

The type of change this PR brings is of so significant  that we might draw a line in the sand here and say use client.* instead?

Reply:
If you say we can safely delete deprecated fields from the api type, I will gladly do so.

Reply:
I think this would be a breaking change that would require bumping up the manifest version as well as the SDK package major version. 
But yes, we can do it 😉

Reply:
Rather not in this PR